helm/dagger/templates/engine-daemonset.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.magicache.enabled }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "dagger.fullname" . }}-magicache-token
+  namespace:  {{ .Release.Namespace }}
+data:
+  _EXPERIMENTAL_DAGGER_CACHESERVICE_TOKEN: {{ required "A magicache token is required" .Values.magicache.token | b64enc | quote }}
+{{- end }}
+{{- if .Values.engine.config }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine-config
+  namespace: {{ .Release.Namespace }}
+data:
+  engine.toml: |{{- .Values.engine.config | nindent 4 }}
+{{- end }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "dagger.fullname" . }}-engine
```

Comment:
Should we use the version in the name - e.g. `dagger-08x` - so that multiple versions can be deployed at the same time?

I am currently running both `v0.6.4` & `v0.8.4` since various pipelines require different Engine versions. Without the name, there can be only one Engine version running in a cluster.

In my case, I keep dedicated nodes for Dagger & ARC runners, and use a `StatefulSet` for Dagger Engine. This is what my current `dagger-08x` `StatefulSet` config looks like:

```yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: dagger-08x
  namespace: actions-runner
spec:
  selector:
    matchLabels:
      name: dagger-08x
  template:
    metadata:
      labels:
        name: dagger-08x
    spec:

      # Schedule on a specific node
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/hostname
                operator: In
                values:
                - par22-w1

      # Match the runner group so that UNIX sockets work
      securityContext:
        runAsUser: 0
        runAsGroup: 1001
        fsGroup: 1001
        fsGroupChangePolicy: "OnRootMismatch"

      containers:
        - name: dagger-engine
          # https://github.com/dagger/dagger/pkgs/container/engine
          image: registry.dagger.io/engine:v0.8.4

          # 🚔Uncomment to debug failures
          # command: ["sleep", "infinity"]

          # https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy
          imagePullPolicy: IfNotPresent

          # Limit the number of build steps to the CPU limit - `num-cpu` does not respect cgroups
          # https://github.com/moby/buildkit/blob/v0.11/cmd/buildkitd/config/config.go#L90-L91
          args:
            - "--oci-max-parallelism"
            - "2"

          ports:
            - containerPort: 8080
              name: grpc

          # All work happens in Dagger, give it as many resources as possible
          resources:
            requests:
              cpu: "2"
              memory: "8Gi"
            limits:
              # cpu: "4"
              memory: 8Gi

          # Dagger Engine requires privileged execution & all capabilities
          securityContext:
            privileged: true
            capabilities:
              add:
                - ALL

          # So that the Dagger client can connect to the Engine
          volumeMounts:
            - name: data
              mountPath: /var/lib/dagger
            - name: socket
              mountPath: /var/run/buildkit

      # Give the Engine plenty of time to shutdown gracefully
      terminationGracePeriodSeconds: 60

      volumes:
        - name: data
          hostPath:
            path: /var/lib/dagger-08x
        - name: socket
          hostPath:
            path: /var/run/dagger-08x
```

Reply:
My preference here would be to just allow the user to override the `name` and `fullname` and the `tolerations` and `affinity` to allow for customization of this.